tools/inspect/endline_whitespace_check.cpp

```diff
@@ -50,7 +53,7 @@ namespace boost
             vector<string> someline, lineorder;
 
 
-            char_separator<char> sep("\n");
+            char_separator<char> sep("\n", "", boost::keep_empty_tokens);
```

Comment:
This is the true culprit, as this allows me to keep the null tokens, so that even if I don't use them, it increases the line number count, so that the line number is actually correct.